internal/elasticsearch/index/index.go

```diff
@@ -716,7 +743,23 @@ func resourceIndexCreate(ctx context.Context, d *schema.ResourceData, meta inter
 		}
 	}
 
-	if diags := elasticsearch.PutIndex(ctx, client, &index); diags.HasError() {
+	params := models.PutIndexParams{
+		WaitForActiveShards: d.Get("wait_for_active_shards").(string),
+		IncludeTypeName:     d.Get("include_type_name").(bool),
```

Comment:
I think we should throw an error if `include_type_name = true` when running against an 8.x server. IMO it avoids potential confusion for users. 

Personally, I would probably do all the version checking in here, switch the `IncludeTypeName` struct property to a pointer and then set it in the create request if it's not nil in the params but I don't feel strongly about it.

Reply:
Fixed to return error when the version is unsupported!

> Personally, I would probably do all the version checking in here, switch the IncludeTypeName struct property to a pointer and then set it in the create request if it's not nil in the params but I don't feel strongly about it.

Agreed, moved the logic here.  We don't know if the value is set or not when zero value (`false`), I simply make it evaluate and set when `include_type_name = true`.